apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/GraphQLCompiler.kt

```diff
@@ -12,11 +12,13 @@ class GraphQLCompiler {
   fun write(args: Arguments) {
     val ir = args.ir
     val customTypeMap = args.customTypeMap.supportedTypeMap(ir.typesUsed)
+    val customSingularizationRules = args.customSingularizationRules.map { Pair(it.key, it.value) }.toHashSet()
```

Comment:
just a curious why not to leave it as Map but convert to hash set? As it looks like with map it will be easier to work later in `String.singularize`?

Reply:
Cause I did a bunch of refactoring and forgot to change it back to a Map! Great catch, thanks.

Reply:
Cause I did a bunch of refactoring and forgot to change it back to a Map! Great catch, thanks.

Reply:
Cause I did a bunch of refactoring and forgot to change it back to a Map! Great catch, thanks.

Reply:
Actually putting to back to a Map made the `singularize` method a bit more confusing, as you have to iterate the keys, then use that to fetch the matching value (which is nullable). Happy to change it if you want, but personally I think the logic for `Set<Pair<String, String>>` ends up being a bit clearer.